spring-core/src/main/java/org/springframework/core/codec/ByteBufDecoder.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.core.codec;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import org.springframework.core.ResolvableType;
+import org.springframework.core.io.buffer.DataBuffer;
+import org.springframework.core.io.buffer.DataBufferUtils;
+import org.springframework.lang.Nullable;
+import org.springframework.util.MimeType;
+import org.springframework.util.MimeTypeUtils;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ * Decoder for {@link ByteBuffer ByteBuffers}.
+ *
+ * @author Vladislav Kisel
+ * @since 5.3
+ */
+public class ByteBufDecoder extends AbstractDataBufferDecoder<ByteBuf> {
+
+	public ByteBufDecoder() {
+		super(MimeTypeUtils.ALL);
+	}
+
+
+	@Override
+	public boolean canDecode(ResolvableType elementType, @Nullable MimeType mimeType) {
+		return (ByteBuf.class.isAssignableFrom(elementType.toClass()) &&
+				super.canDecode(elementType, mimeType));
+	}
+
+	@Override
+	public ByteBuf decode(DataBuffer dataBuffer, ResolvableType elementType,
+			@Nullable MimeType mimeType, @Nullable Map<String, Object> hints) {
+
+		int byteCount = dataBuffer.readableByteCount();
+		ByteBuf copy = Unpooled.buffer(byteCount);
+		copy.writeBytes(dataBuffer.asByteBuffer());
+		copy.readerIndex(0);
+		DataBufferUtils.release(dataBuffer);
```

Comment:
If running on Netty, the copy here is unnecessarily inefficient. If `DataBuffer` is already an instance of `NettyDataBuffer` then we should simply return `((NettyDataBuffer) dataBuffer).getNativeBuffer()`.